osu.Game/Screens/Menu/IntroTriangles.cs

```diff
@@ -0,0 +1,413 @@
+    {
+        private const string menu_music_beatmap_hash = "a1556d0801b3a6b175dda32ef546f0ec812b400499f575c44fccbe9c67f9b1e5";
+
+        private SampleChannel welcome;
+
+        protected override BackgroundScreen CreateBackground() => background = new BackgroundScreenDefault(false)
+        {
+            Alpha = 0,
+        };
+
+        [Resolved]
+        private AudioManager audio { get; set; }
+
+        private Bindable<bool> menuMusic;
+        private Track track;
+        private WorkingBeatmap introBeatmap;
+
+        private BackgroundScreenDefault background;
+
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config, BeatmapManager beatmaps, Framework.Game game)
+        {
+            menuMusic = config.GetBindable<bool>(OsuSetting.MenuMusic);
+
+            BeatmapSetInfo setInfo = null;
+
+            if (!menuMusic.Value)
+            {
+                var sets = beatmaps.GetAllUsableBeatmapSets();
+                if (sets.Count > 0)
+                    setInfo = beatmaps.QueryBeatmapSet(s => s.ID == sets[RNG.Next(0, sets.Count - 1)].ID);
+            }
+
+            if (setInfo == null)
+            {
+                setInfo = beatmaps.QueryBeatmapSet(b => b.Hash == menu_music_beatmap_hash);
+
+                if (setInfo == null)
+                {
+                    // we need to import the default menu background beatmap
+                    setInfo = beatmaps.Import(new ZipArchiveReader(game.Resources.GetStream(@"Tracks/triangles.osz"), "triangles.osz")).Result;
+
+                    setInfo.Protected = true;
+                    beatmaps.Update(setInfo);
+                }
+            }
+
+            introBeatmap = beatmaps.GetWorkingBeatmap(setInfo.Beatmaps[0]);
+
+            track = introBeatmap.Track;
+            track.Reset();
+
+            if (config.Get<bool>(OsuSetting.MenuVoice) && !menuMusic.Value)
+                // triangles has welcome sound included in the track. only play this if the user doesn't want menu music.
+                welcome = audio.Samples.Get(@"welcome");
+        }
+
+        protected override void LogoArriving(OsuLogo logo, bool resuming)
+        {
+            base.LogoArriving(logo, resuming);
+
+            logo.Triangles = true;
+
+            if (!resuming)
+            {
+                Beatmap.Value = introBeatmap;
+                introBeatmap = null;
+
+                PrepareMenuLoad();
+
+                LoadComponentAsync(new TrianglesIntroSequence(logo, background)
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Clock = new FramedClock(track),
+                    LoadMenu = LoadMenu
+                }, t =>
+                {
+                    AddInternal(t);
+                    welcome?.Play();
+
+                    // Only start the current track if it is the menu music. A beatmap's track is started when entering the Main Manu.
+                    if (menuMusic.Value)
```

Comment:
This line will cause the game to get stuck on the intro screen if the user has disabled the menu music at startup. Commenting / removing it seems to fix the problem.